packages/client/src/app/pages/my-stuff/section-item/section-item.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, Input, OnInit, Output, EventEmitter, OnChanges } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+import { SectionsService } from '../../../services/user';
+import { PublishSection, SectionForm } from '@pulp-fiction/models/sections';
+import { SectionItem } from '../viewmodels';
+
+@Component({
+    selector: 'section-item',
+    templateUrl: './section-item.component.html',
+    styleUrls: ['./section-item.component.less']
+})
+export class SectionItemComponent implements OnInit, OnChanges {
+    @Input() contentId: string;
+    @Input() section: SectionItem;
+    @Input() selected: boolean;
+    @Output() selectItem = new EventEmitter<boolean>();
+
+    previewMode = true;
+
+    editForm = new FormGroup({
+        title: new FormControl('', [Validators.required, Validators.minLength(3), Validators.maxLength(100)]),
+        body: new FormControl('', [Validators.required, Validators.minLength(3)]),
+        authorsNote: new FormControl('', [Validators.minLength(3)]),
+        authorsNotePos: new FormControl(null)
+    });
+
+    constructor(private snackBar: MatSnackBar, private sectionsService: SectionsService) {}
+
+    get fields() { return this.editForm.controls; }
+
+    ngOnInit(): void {
+        this.editForm.setValue({
+            title: this.section.title,
+            body: this.section.body,
+            authorsNote: this.section.authorsNote,
+            authorsNotePos: this.section.authorsNotePos
+        });
+    }
+
+    ngOnChanges(): void {
+        this.editForm.setValue({
+            title: this.section.title,
+            body: this.section.body,
+            authorsNote: this.section.authorsNote,
+            authorsNotePos: this.section.authorsNotePos
+        });
+    }
```

Comment:
Am I misunderstanding how the ngOnChanges event works, or wouldn't this prevent any editing from happening?